src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -4053,12 +4053,12 @@ private List<JCTree> constructUnnamedClass(List<JCTree> origDefs) {
         }
 
         Name name = names.fromString(simplename);
-        JCModifiers unnamedMods = F.at(Position.NOPOS)
-                .Modifiers(Flags.FINAL|Flags.SYNTHETIC|Flags.UNNAMED_CLASS, List.nil());
-        JCClassDecl unnamed = F.at(primaryPos).ClassDef(
-                unnamedMods, name, List.nil(), null, List.nil(), List.nil(),
+        JCModifiers implicitMods = F.at(Position.NOPOS)
+                .Modifiers(Flags.FINAL|Flags.MANDATED|Flags.IMPLICIT_CLASS, List.nil());
```

Comment:
I don't think the JVMS allows for a flag to have ACC_MANDATED flags?

```
It is a limitation of the class file format that only formal parameters and modules can be flagged as ACC_MANDATED ([§4.7.24](https://docs.oracle.com/javase/specs/jvms/se21/html/jvms-4.html#jvms-4.7.24), [§4.7.25](https://docs.oracle.com/javase/specs/jvms/se21/html/jvms-4.html#jvms-4.7.25)) to indicate that, despite being compiler-generated, they are not considered implementation artifacts. There is no way to flag other compiler-generated constructs so that they too are not considered implementation artifacts (JLS §13.1). This limitation means that reflective APIs of the Java SE Platform may not accurately indicate the "mandated" status of such constructs.
```

Reply:
The spec has been updated to include implicit classes.

Reply:
Uhmm. In the spec I see implicit classes named here:

```
 For reference, the following constructs are declared implicitly in source code, but are not marked as mandated because only formal parameters and modules can be so marked in a class file (JVMS §4.7.24, JVMS §4.7.25):
 ```

Note the `but are not marked as mandated`.

I don't think javac sets the mandated flag for any of these. Note that ACC_MANDATED has the same value as the ACC_MODULE flag, so setting that flag on a class w/o at least filtering out when writing the classfile will be problematic.

Reply:
Hmmm. I'll check with Gavin, but I thought in discussions Alex said that it should be mandated. At any rate, seems the ACC_MANDATED should be removed for stated reasons.

Reply:
It should be MANDATED in theory given it's not an artifact of the compilation like SYNTHETIC is. Sadly, as Maurizio said ACC_MANDATED has the same value as ACC_MODULE.
I think there is a value to still make it as SYNTHETIC, apart if we add a new specific attribute for implicit classes.

Reply:
We're opting to not to flag the class as special. Implicit classes are just regular classes derived from syntactic sugar.